src/kernels/raw/launcher/kernelEnvVarsService.node.ts

```diff
@@ -37,13 +44,7 @@ export class KernelEnvironmentVariablesService {
         let kernelEnv = kernelSpec.env && Object.keys(kernelSpec.env).length > 0 ? kernelSpec.env : undefined;
 
         // If an interpreter was not explicitly passed in, check for an interpreter path in the kernelspec to use
-        if (!interpreter) {
-            if (!kernelSpec.interpreterPath) {
-                traceInfo(
-                    `No custom variables for Kernel as interpreter path is not defined for kernel ${kernelSpec.display_name}`
-                );
-                return kernelEnv;
```

Comment:
We will always return the env variables in this method, never `undefined`
i.e. merge process.env with kernelspec.env and the like.